src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -2660,7 +2662,19 @@ MultiRouterPlannableQuery(Query *query, RelationRestrictionContext *restrictionC
 			char partitionMethod = PartitionMethod(distributedTableId);
 
 			if (!(partitionMethod == DISTRIBUTE_BY_HASH || partitionMethod ==
-				  DISTRIBUTE_BY_NONE))
+				  DISTRIBUTE_BY_NONE || partitionMethod == DISTRIBUTE_BY_RANGE))
+			{
+				return false;
+			}
+
+			hasHashDistributedTable |= partitionMethod == DISTRIBUTE_BY_HASH;
```

Comment:
Do we really need this? Shard pruning logic already handles restrictions differently per distribution model. Even if we use hash and range partitioned tables in the same query, they both should be able to pruned down to a single shard.

I don't see a reason for this check also since we allow such queries on multi-shard planner. For instance, if you enable task-tracker on the test that errors out, the query yields the correct result.